share/eds/eds.go

```diff
@@ -293,5 +296,47 @@ func ProveShares(eds *rsmt2d.ExtendedDataSquare, start, end int) (*types.SharePr
 	if err != nil {
 		return nil, err
 	}
-	return &proof, nil
+	coreProof := toCoreShareProof(proof)
+	return &coreProof, nil
+}
+
+// toCoreShareProof utility function that converts a share proof defined in app
+// to the share proof defined in node.
+// This will be removed once we unify both these proofs.
+// Reference issue: https://github.com/celestiaorg/celestia-app/issues/3734
+func toCoreShareProof(appShareProof pkgproof.ShareProof) types.ShareProof {
+	shareProofs := make([]*coretypes.NMTProof, 0)
```

Comment:
Thanks! Resolved via f52886c5c0ea8a76a41d8ad056494451e0220f29

Reply:
I had to revert it in 35f5164 because it looks like `len(appShareProof.ShareProofs)` or `len(appShareProof.RowProof.Proofs)` can be nil so we don't know what size to allocate for the slice.